src/alertsmanagement/azext_alertsmanagement/custom.py

```diff
@@ -0,0 +1,324 @@
+    if monitor_condition is not None:
+        properties['conditions']['monitorCondition'] = monitor_condition
+    if target_resource_type is not None:
+        properties['conditions']['targetResourceType'] = target_resource_type
+    if alert_rule_id is not None:
+        properties['conditions']['alertRuleId'] = alert_rule_id
+    if alert_description is not None:
+        properties['conditions']['description'] = alert_description
+    if alert_context is not None:
+        properties['conditions']['alertContext'] = alert_context
+    properties['suppressionConfig'] = {
+        'recurrenceType': recurrence_type,
+        'schedule': {
+            'startDate': start_date,
+            'endDate': end_date,
+            'startTime': start_time,
+            'endTime': end_time
+        }
+    }
+    body['properties'] = properties
+
+    return client.create_update(resource_group_name=resource_group, action_rule_name=name, action_rule=body)
+
+
+def update_alertsmanagement_action_rule(instance, client,
+                                        location=None,
+                                        tags=None,
+                                        status=None):
+    if location is not None:
+        instance.location = location
+    if tags is not None:
+        instance.tags = tags
+    if status is not None:
+        instance.properties.status = status
+    return instance
+
+
+def delete_alertsmanagement_action_rule(cmd, client,
+                                        resource_group,
+                                        name):
+    return client.delete(resource_group_name=resource_group, action_rule_name=name)
+
+
+def get_alertsmanagement_action_rule(cmd, client,
+                                     resource_group,
+                                     name):
+    return client.get_by_name(resource_group_name=resource_group, action_rule_name=name)
+
+
+def list_alertsmanagement_action_rule(cmd, client,
+                                      resource_group=None,
+                                      target_resource_group=None,
+                                      target_resource_type=None,
+                                      target_resource=None,
+                                      severity=None,
+                                      monitor_service=None,
+                                      impacted_scope=None,
+                                      description=None,
+                                      alert_rule_id=None,
+                                      action_group=None,
+                                      name=None):
+    if resource_group is not None and target_resource_group is not None and target_resource_type is not None and target_resource is not None and severity is not None and monitor_service is not None and impacted_scope is not None and description is not None and alert_rule_id is not None and action_group is not None and name is not None:
```

Comment:
format by wrapping lines for readability :)

Reply:
we need to provide so many parameters to do a list_by_resource_group?

Reply:
> 
> 
> format by wrapping lines for readability :)

Done

Reply:
> 
> 
> we need to provide so many parameters to do a list_by_resource_group?

Don't worry. They are optional.